lib/reporting/reports/revenues_by_hub/base.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Reporting
+  module Reports
+    module RevenuesByHub
+      class Base < ReportTemplate
+        def search
+          permissions = ::Permissions::Order.new(user)
+          permissions.editable_orders.complete.not_state(:canceled).ransack(ransack_params)
+        end
+
+        def default_params
+          {
+            q: {
+              completed_at_gt: 1.month.ago.beginning_of_day,
+              completed_at_lt: 1.day.from_now.beginning_of_day
+            }
+          }
+        end
+
+        def columns
+          {
+            hub: proc { |orders| distributor(orders).name },
+            hub_id: proc { |orders| distributor(orders).id },
```

Comment:
The CSV export we use in Australia has a few more columns. We could probably fetch them separately but if it could all be included in the same report then it would be super easy to switch to this new report.

```
SELECT
  distributor_id,
  abn,
  enterprises.name,
  business_address.company legal_name,
  enterprises.contact_name,
  enterprises.email_address,
  enterprises.phone,
  replace(address.address1, ',', ' ') address_line1,
  replace(address.address2, ',', ' ') address_line2,
  address.city,
  address.zipcode,
  address.state_name,
  count(spree_orders.id) orders,
  sum(spree_orders.total) distributor_total
```

The whole thing we use:
https://gist.github.com/mkllnk/9be5e677bffb65141998d9ff9925ecc1

I'm not sure about the funding of this feature. @kirstenalarsen are we happy to pay my time to add to this PR for AU specific additions?

There's no issue linked to this PR. I just created a Clockify project to track my review time on this as funded feature. It's called "Revenue report".

Reply:
> There's no issue linked to this PR
Yes sorry. I just had the time to make [this slack post](https://openfoodnetwork.slack.com/archives/C03HE00FSA2/p1659538858780589), after which JB quickly did the PR... as I planned to test it comparing the file I'm doing manually, I thought it wouldn't need an issue, but I forgot code review, sorry about that.

Reply:
would it be any issue to anyone else if we added these fields? i.e. clogging anything up? If not, I think probably we should do it and then we can just drop maintenance / use of our separate script. I'm just checking with the team - TBC